### ISSUE TITLE
Multiple source sentences

### ISSUE THREAD

**Issue by pmichel31415** [neulab/xnmt#347]
Here I go asking for things again.

It would be nice to have a way to condition on multiple input sentences and ensemble the translations with either the same model or a different model.

What would be the easiest/most straightforward way for me to add this functionality to xnmt.

I'm think of being able to specify multiple source files at inference (assuming they are aligned and have the same number of sentences), a list of models corresponding to each input file, with references if I want to use the same model, and optionally an ensembilng method (eg average logits vs average log probabilities)
